control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -499,13 +500,15 @@ func (r *reconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result
 	// so we must use the uncached client for this delete call.  To avoid
 	// excessive API calls using the uncached client, the delete call is
 	// guarded using a sync.Once.
-	deleteDNSOperatorDeploymentOnce.Do(func() {
-		dnsOperatorDeployment := manifests.DNSOperatorDeployment()
-		log.Info("removing any existing DNS operator deployment")
-		if err := r.uncachedClient.Delete(ctx, dnsOperatorDeployment); err != nil && !apierrors.IsNotFound(err) {
-			errs = append(errs, err)
-		}
-	})
+	if r.isClusterVersionUpdated(ctx, releaseImage.Version()) {
+		deleteDNSOperatorDeploymentOnce.Do(func() {
+			dnsOperatorDeployment := manifests.DNSOperatorDeployment()
+			log.Info("removing any existing DNS operator deployment")
+			if err := r.uncachedClient.Delete(ctx, dnsOperatorDeployment); err != nil && !apierrors.IsNotFound(err) {
+				errs = append(errs, err)
+			}
+		})
+	}
```

Comment:
@a-dsouza can you add the block below (`deleteCVORemovedResourcesOnce.Do`) under the same `if`? The old CVO will also attempt to recreate those resources.

Reply:
Yup, just pushed the changes--thanks!